File: backend/app/schemas/signaling.py
from typing import Optional, Any, List
from pydantic import BaseModel

class DeviceInfo(BaseModel):
    peer_id: str
    device_name: str
    device_type: str

class SignalMessage(BaseModel):
    type: str
    from_peer: Optional[str] = None
    to_peer: Optional[str] = None
    payload: Optional[Any] = None
    peer: Optional[DeviceInfo] = None
    peer_id: Optional[str] = None
    peers: Optional[List[DeviceInfo]] = None

class PeersListResponse(BaseModel):
    peers: List[DeviceInfo]

File: frontend/src/types/signaling.ts
export interface DeviceInfo {
  peer_id: string;
  device_name: string;
  device_type: 'desktop' | 'mobile' | 'tablet';
}

export interface SignalMessage {
  type: 'offer' | 'answer' | 'ice-candidate' | 'peer-joined' | 'peer-left' | 'peer-list' | 'chat' | 'clipboard' | 'ping' | 'pong';
  from_peer?: string;
  to_peer?: string;
  payload?: unknown;
  peer?: DeviceInfo;
  peer_id?: string;
  peers?: DeviceInfo[];
}
